packages/@aws-cdk/aws-appmesh/lib/gateway-route.ts

```diff
@@ -0,0 +1,390 @@
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnGatewayRoute } from './appmesh.generated';
+import { Protocol } from './shared-interfaces';
+import { IVirtualGateway, VirtualGateway } from './virtual-gateway';
+import { IVirtualService } from './virtual-service';
+
+/**
+ * Interface for which all GatewayRoute based classes MUST implement
+ */
+export interface IGatewayRoute extends cdk.IResource {
+  /**
+   * The name of the GatewayRoute
+   *
+   * @attribute
+   */
+  readonly gatewayRouteName: string,
+
+  /**
+   * The Amazon Resource Name (ARN) for the GatewayRoute
+   *
+   * @attribute
+   */
+  readonly gatewayRouteArn: string;
+
+  /**
+   * The VirtualGateway the GatewayRoute belongs to
+   *
+   * @attribute
```

Comment:
I *think* `@attribute` should not be here...? (These are only for CloudFormation attributes)

Reply:
I was following the other resources which have them present in the other `IResource` interface.

Looks like lambda is doing this too: https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda/lib/function-base.ts#L19

Reply:
Right. Lambda is doing it for `functionName`, which _is_ an attribute 🙂.

I don't have any problems with `gatewayRouteName` and `gatewayRouteArn` being `@attribute`s in `IGatewayRoute`. I just don't think `virtualGateway` should be one.

Reply:
ahh got it. I thought you meant having the `attribute` tag in general

Reply:
`@attribute` is still here 🙂🙂

Reply:
my bad!